src/testing/tmp_tigerbeetle.zig

```diff
@@ -106,12 +118,58 @@ pub fn init(
         .port = port,
         .port_str = port_str,
         .tmp_dir = tmp_dir,
+        .stderr_reader = reader_maybe,
         .process = process,
     };
 }
 
-pub fn deinit(tb: *TmpTigerBeetle, gpa: std.mem.Allocator) void {
-    _ = gpa;
-    _ = tb.process.kill() catch unreachable;
+pub fn deinit(tb: *TmpTigerBeetle) void {
+    if (tb.stderr_reader) |reader| {
+        _ = tb.process.kill() catch unreachable;
```

Comment:
Due to the pattern of 

```
defer deinit()
errdefer log_stderr()
```

both functions can be called at the same time. To log stderr detreminstically, we need to kill the process and join the thread (an alternative would be to add a mutex and log "what we have so far", but that would give non-deterministic result). And it looks like
while `process.kill` is idempotent, `thread.join` is not (its UB to cal it twice). 

So I have to add my own idempotency on top, by using `tb.stderr_reader != null` condition to check the joining was done already. 


In genreal, the synchronization here is very tricky! I don't love this code, but I don't see anything substantionally simpler (what I'd _love_ to do is to just dump stderr to file, but Zig doesn't allowing setting child's stderr to a file descriptor).

Reply:
Ahh I see. `tb.stderr_reader` is always non-null until either `deinit()` or `log_stderr` is called -- whichever one executes first also nullifies `stderr_reader` so that we don't try to kill the thread twice.